packages/multichain-account-service/src/MultichainAccountWallet.ts

```diff
@@ -173,4 +173,69 @@ export class MultichainAccountWallet<
   getMultichainAccountGroups(): MultichainAccountGroup<Account>[] {
     return Array.from(this.#accounts.values()); // TODO: Prevent copy here.
   }
+
+  /**
+   * Gets next group index for this wallet.
+   *
+   * @returns The next group index of this wallet.
+   */
+  getNextGroupIndex(): number {
+    // Assuming we cannot have indexes gaps.
+    return this.#accounts.size; // No +1 here, group indexes starts at 0.
+  }
+
+  #createMultichainAccountGroup(
+    groupIndex: number,
+  ): MultichainAccountGroup<Account> {
+    const multichainAccount = new MultichainAccountGroup({
+      wallet: this,
+      providers: this.#providers,
+      groupIndex,
+    });
+
+    // Register the account to our internal map.
+    this.#accounts.set(groupIndex, multichainAccount);
+
+    return multichainAccount;
+  }
+
+  async createMultichainAccountGroup(
+    groupIndex: number,
+  ): Promise<MultichainAccountGroup<Account>> {
+    const nextGroupIndex = this.getNextGroupIndex();
+    if (groupIndex > nextGroupIndex) {
+      throw new Error(
+        `You cannot use a group index that is higher than the next available one: expected <=${nextGroupIndex}, got ${groupIndex}`,
+      );
+    }
+
+    const group = this.getMultichainAccountGroup(groupIndex);
+    if (group) {
+      // If the group already exists, we just `sync` it and returns the same
+      // reference.
+      group.sync();
+
+      return group;
+    }
+
+    // TODO: Make this parallel.
+    for (const provider of this.#providers) {
+      // FIXME: What to do if any provider fails to create accounts?
+      await provider.createAccounts({
+        entropySource: this.#entropySource,
+        groupIndex,
+      });
+    }
+
+    // Re-create and "refresh" the multichain account (we assume all account creations are
+    // idempotent, so we should get the same accounts and potentially some new accounts (if
+    // some account providers decide to return more of them this time).
```

Comment:
Not sure if I understand this comment, if we're at this point wouldn't the accounts created from this group all be "new"?

Reply:
Yes, actually this is an old comment that does not make sense anymore. I did remove it.

Basically, a multichain account group could be "partially" composed of BIP-44 accounts, like, it could have 1 EVM account, but no Solana account yet. This is true only for the moment because we're migrating our current account model.

Before I introduced the `sync` method, I was generally "re-building" the same group again to let it re-populate its internal list of accounts, but now we can just use `sync` to re-sync with the account providers.

Lastly, even after re-sync we might still miss a Solana account (in my previous example). This issue will be fixed once we introduce the "alignment mechanism" which will automatically create missing account for a given wallet and group index combo.